src/login/logind-user.c

```diff
@@ -382,60 +369,50 @@ int user_start(User *u) {
         return 0;
 }
 
-static int user_stop_slice(User *u) {
+static void user_stop_service(User *u) {
         _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
-        char *job;
         int r;
 
         assert(u);
+        assert(u->service);
 
-        r = manager_stop_unit(u->manager, u->slice, &error, &job);
-        if (r < 0)
-                return log_error_errno(r, "Failed to stop user slice: %s", bus_error_message(&error, r));
+        /* The reverse of user_start_service(). Note that we only stop user@UID.service here, and let StopWhenUnneeded=
+         * deal with the slice and the user-runtime-dir@.service instance. */
```

Comment:
Are you sure? I know user-runtime-dir@.service has StopWhenUnneeded=yes. But I don't know where user-UID.slice is defined, to set StopWhenUnneeded=yes.

Reply:
It is added in https://github.com/systemd/systemd/pull/9824/commits/25913f1b317b8cbe058c4d4272d75dc1495932f2.